feat: app usage stats

cimigree opened the pull request:
closes #1284 

## Add App Usage Prompt Interstitial + Tests

### Summary

This PR introduces a new  screen that prompts users to share how they use CoMapeo. The screen is conditionally shown based on onboarding and prompt timing logic, and is meant to support opt-in usage data collection.

### Key Features

* **New Screen**: `AppUsagePromptInterstitial`

  * Displays a privacy-first usage opt-in prompt
  * Automatically appears 7 days after onboarding and then again after 3 months if not opted in previously
  * Respects user dismissal and tracks `promptCount` and `lastPromptAt` state

* **State Management**:

  * Backed by `useAppUsageStatsPromptStore` (Zustand)
  * Logic encapsulated in `shouldShowAppUsagePrompt(...)`

* **Navigation Integration**:

  * Logic lives in the RootStack for when to show (or not show) the screen
  * Automatically skips the screen if the prompt should not be shown

* **Testing**:

  * Unit tests for logic
  * Tests for Context
  * My first integration test! That was a bear!

cimigree committed:
First stab at setting up context for app usage stats prompt

cimigree committed:
Adds logic for when to show interstitial. Adds interstitial. Adds navigation for interstitial.

cimigree committed:
Messages for app usage prompt

cimigree committed:
Adds success screen.

cimigree committed:
Updates logic to show success screen. Updates messages. Updates navigation for typescript.

cimigree committed:
Adds unit tests

cimigree committed:
Merges with develop branch

cimigree committed:
Adds prompt store to the test helper

cimigree committed:
Merges with develop

cimigree committed:
Test that sort of works. Issues with animations

cimigree committed:
Adds more tests. Some cleanup

cimigree committed:
Fixes redundant navigation

ErikSin commented on src/frontend/contexts/AppUsageStatsPromptContext.tsx:
```diff
@@ -0,0 +1,87 @@
+  completedOnboardingAt: number | null;
+  lastPromptAt: number | null;
+  promptCount: number;
+}
```
We should use valibot as it makes migrations much simpler

cimigree replied:
https://github.com/digidem/comapeo-mobile/pull/1312/commits/afcecf5fc76487ff038257a91009544d94cf1b5a

ErikSin commented on src/frontend/screens/AppUsagePromptInterstitial.tsx:
```diff
@@ -0,0 +1,162 @@
+    if (!showPrompt) {
+      navigation.replace('Home', {screen: 'Map'});
+    }
+  }, [showPrompt, navigation]);
```
when screens are being conditionally rendered, [we do not want to manually navigate](https://reactnavigation.org/docs/auth-flow/#dont-manually-navigate-when-conditionally-rendering-screens)

cimigree replied:
https://github.com/digidem/comapeo-mobile/pull/1312/commits/afcecf5fc76487ff038257a91009544d94cf1b5a

ErikSin commented on src/frontend/screens/AppUsageSharingSuccess.tsx:
```diff
@@ -0,0 +1,92 @@
+    if (optedIn) {
+      navigation.replace('Home', {screen: 'Map'});
+    }
+  }, [optedIn, navigation]);
```
Same comment as above, dont navigate when conditionally rendering

cimigree replied:
https://github.com/digidem/comapeo-mobile/pull/1312/commits/afcecf5fc76487ff038257a91009544d94cf1b5a

ErikSin commented on src/frontend/screens/AppUsagePrompt.test.tsx:
```diff
@@ -0,0 +1,178 @@
+    });
+
+    render(
+      <NavigationContainer>
+        <AppUsageStatsPromptProvider value={appUsageStore}>
+          <RootStack.Navigator>
+            <RootStack.Screen
+              name="AppUsagePromptInterstitial"
+              component={AppUsagePromptInterstitial}
+              options={{headerShown: false}}
+            />
+            <RootStack.Screen
+              name="AppUsageSharingSuccess"
+              component={AppUsageSharingSuccess}
+              options={{headerShown: false}}
+            />
+            <RootStack.Screen
+              name="Home"
+              component={HomeTabs}
+              options={{headerShown: false}}
+            />
+          </RootStack.Navigator>
+        </AppUsageStatsPromptProvider>
+      </NavigationContainer>,
```
Your tests are not properly testing the actual use case. In the actual nav stack, we are conditionally rendering these screens based on the `AppUsageStat` state . Here you are ALWAYS rendering the screens. I think the best thing would be to mount the entire AppStack and set the `AppUsageStatsPromptProvider` state (similar to how your doing it, but [directly using the hooks](https://github.com/testing-library/react-hooks-testing-library) in your tests.

ErikSin replied:
I think skip these test for now, and lets create a ticket to address it later. I think i can update the testing environment to make this easier in the future. I ran some experiments and it just not working the way they should be

cimigree replied:
https://github.com/digidem/comapeo-mobile/pull/1312/commits/afcecf5fc76487ff038257a91009544d94cf1b5a

ErikSin reviewed (changes requested):
We are missing the use case where the use upgrades the app and already has done onboarding. If that's the case, `completedOnboardingAt` will always be null.

The suggest fix is to run a useEffect. In that useEffect check for deviceName. And if there is a deviceName AND completedOnboardingAt is null, set it completedOnboardingAt to Date.now() - So it will run 7 days after the user has upgraded

cimigree committed:
Adds optInStarted to comply with specs. Removes test that did not do the right things. Updated conditional navigation. Uses valibot. Addresses case for existing users.

cimigree commented:
I added the useEffect in the RootStack. That file will need some adjustment when the other PR gets integrated...
I realized that I missed a requirement. This one: 

> if tapped "Yes, count me in" gather data for up to 12 months at which point the feature gets turned off

So I added an opted in date (and expiry date) to the storage. Not sure if I should have included the expiry date. I did for now but can see calculating that on the fly. I wasn't sure what was the best way.

cimigree committed:
Removes test code. Oops

cimigree committed:
Merges with develop

cimigree committed:
Resolves merge conflicts. Merges with develop.

cimigree committed:
Merges with develop

cimigree committed:
Fixes missing value.

ErikSin commented:
Can you update from develop and take out these screens from navigation (aka make them inaccessible to the user)? I am going to use these screen eventually, but instead of maintaing this pr, lets just merge it and not make it accessible to the user but allow us to add these screens really easily when needed

ErikSin commented:
Can we also take out all uses of the AppUsagePromtContext. Again Im going to use it eventually, but for now lets just hide until i can properly connect it to posthog

cimigree committed:
Merges with develop. Removes screen visibility.

cimigree committed:
Removes all app usage stats stuff from view and use.

cimigree committed:
Adds back navigation types. removes navigation from interstitial

cimigree commented:
@ErikSin , I think I removed all usages as requested. What do you think?

ErikSin reviewed (approved):
Thank you
